### PR TITLE
"Show on" dropdown

### PR DESCRIPTION
**Pull request by ngfeldman**
Added global setting to set whether social plugins show on all posts, all pages, both, or neither. Done for like button, subscribe button, send button, comments, and recommendations bar

### CODE REVIEW

**Comment by mattwkelly** [comment]
Looking good.
- We don't need a 'none' option, though.  None is equivalent as disabling the entire option.
- 'all posts and pages' should be the default for everything.

File: social-plugins/fb-comments.php [hunk 1]
```diff
@@ -46,16 +46,19 @@ function fb_comments_automatic($content) {
...
-	
-				foreach($options['comments'] as $param => $val) {
-					$param = str_replace('_', '-', $param);
-	
-					$params[$param] = $val;
-				}
-	
-				$content .= fb_get_comments($params);
+			$options = get_option('fb_options');
+			if (!is_home() && $options['comments']['show_on']) {
```
Comment (mattwkelly) [inline social-plugins/fb-comments.php]: See http://codex.wordpress.org/WordPress_Coding_Standards

"Put spaces on both sides of the opening and closing parenthesis of if, elseif, foreach, for and switch blocks."

File: social-plugins/fb-comments.php [hunk 2]
```diff
@@ -46,16 +46,19 @@ function fb_comments_automatic($content) {
...
+				if ( ( is_page() && ( $options['comments']['show_on']=='all pages' || $options['comments']['show_on'] == 'all posts and pages' ) )
```
Comment (mattwkelly) [inline social-plugins/fb-comments.php]: Some small formatting issues (['show_on']=='all pages' should be ['show_on'] == 'all pages').

There are a couple of similar issues below, too.

File: social-plugins/fb-comments.php [hunk 3]
```diff
@@ -136,6 +139,12 @@ function fb_get_comments_fields_array() {
 													'options' => array('light' => 'light', 'dark' => 'dark'),
 													'help_text' => 'The color scheme of the plugin.',
 													),
+										array('name' => 'show_on',
+													'type' => 'dropdown',
+													'default' => 'all posts',
+													'options' => array('all posts' => 'all posts', 'all pages' => 'all pages', 'all posts and pages' => 'all posts and pages', 'none' => 'none'),
+													'help_text' => __( 'Changes whether the plugin appears on all posts or pages. When changed, individual settings are removed.', 'facebook' ),
```
Comment (mattwkelly) [inline social-plugins/fb-comments.php]: "Changes whether the plugin appears on all posts or pages. When changed, individual settings are removed."

What do you mean by, "When changed, individual settings are removed."?

**Comment by mattwkelly** [comment]
Btw, you can just commit changes to your branch/master, and this issue will auto-update.

**Comment by ngfeldman** [comment]
Technically if 'none' is selected then the plugin could still show on the homepage and nowhere else. Should I still get rid of it?

**Comment by mattwkelly** [comment]
That would be confusing though, given that admins would expect that drop-down to apply posts and pages on the homepage.

Let's remove it for now.

**Comment by ngfeldman** [comment]
And by "When changed..." I mean that when this setting is changed, the individual setting for that plugin to show up any particular page or post would be overwritten, for instance, if it's being changed from all pages to all posts and pages, it wouldn't matter which posts were selected to show or not show the plugin, now they would all show it.

I'll get rid of that sentence for now until I put in the individual settings. I just thought of a case where it might not have made sense even if I did say it well, so the way this should work should probably be ironed out with you anyway.

File: social-plugins/fb-comments.php (hunk 1 above)
Comment (ngfeldman) [inline social-plugins/fb-comments.php]: fiiiiine :p

I was sort of following those as I typed but I wasn't always keeping them in mind.

Comment (mattwkelly) [inline social-plugins/fb-comments.php]: Np!  I'm not a fan of them, honestly, but we need to follow WP standards.

**Comment by mattwkelly** [comment]
We shouldn't override per-post/page settings. I think it will be too confusing.

Long term, maybe we can pop a warning that says "do you want to retroactively change per-post/page settings, too?" or something similar.  For now, I think this is a good v1, and then we can see what kind of feedback people have.

File: social-plugins/fb-comments.php (hunk 1 above)
Comment (ngfeldman) [inline social-plugins/fb-comments.php]: should i bother making the ( $options['like']['show_on'] == 'all pages' )'s backwards for yoda too?

**Comment by ngfeldman** [comment]
is there a way to merge this with issue 145?

**Comment by mattwkelly** [comment]
Why? Let's just keep them separate and I'll merge this pull request (reviewing now).

**Pull request merged**